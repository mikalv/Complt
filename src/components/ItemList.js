import React from 'react';
import { List } from 'material-ui/List';
import Item from './Item';
import OakPropTypes from '../PropTypes';

const ItemList = ({ items, onItemAvatarTap }) => (
  <List>
    {items.map((item, i) =>
      <Item id={i} key={i} item={item} onAvatarTouchTap={() => onItemAvatarTap(i)} />)}
  </List>
);

ItemList.propTypes = {
  items: React.PropTypes.arrayOf(OakPropTypes.item),
  onItemAvatarTap: React.PropTypes.func,
};

export default ItemList;
