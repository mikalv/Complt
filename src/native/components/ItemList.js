import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from '../../common/PropTypes';
import Item from './Item';


const List = props => (
  <ScrollView>
    {props.items.map((item, i) => {
      if (!item) return null;
      return (<Item
        key={item._id}
        item={item}
        onAvatarPress={props.onAvatarPress !== undefined ? () => props.onAvatarPress(i) : undefined}
        onItemPress={props.onAvatarPress !== undefined ? () => props.onAvatarPress(i) : undefined}
        onDeletePress={props.onDeletePress !== undefined ? () => props.onDeletePress(i) : undefined}
      />);
    })}
  </ScrollView>
);

List.propTypes = {
  items: React.PropTypes.arrayOf(PropTypes.item),
  onAvatarPress: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onItemPress: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onDeletePress: React.PropTypes.func,
};

export default List;
