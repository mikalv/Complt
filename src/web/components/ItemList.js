import React from 'react';
import Item from './Item';
import PropTypes from '../../common/PropTypes';

const ItemList = ({
    items = [],
    onItemAvatarTap,
    style,
    onDelete,
    canDeleteTask,
    canDeleteProject,
    canMove,
    onItemTap,
    onItemUpdate,
    onItemMove,
    className,
}) => (
  <div style={style} className={className}>
    {items.map((item, i) => {
      if (!item) return null;
      return (<Item
        key={item._id}
        item={item}
        canDelete={
          (canDeleteTask && item.isProject === false) ||
          (canDeleteProject && item.isProject === true)}
        canMove={canMove}
        onAvatarTouchTap={onItemAvatarTap !== undefined ? () => onItemAvatarTap(i) : undefined}
        onItemTap={onItemTap !== undefined ? () => onItemTap(i) : undefined}
        onItemUpdate={onItemUpdate !== undefined ? () => onItemUpdate(i) : undefined}
        onItemMove={onItemMove !== undefined ? () => onItemMove(i) : undefined}
        onDelete={() => onDelete(i)}
      />);
    })}
  </div>
);

ItemList.propTypes = {
  items: React.PropTypes.arrayOf(PropTypes.item),
  onItemAvatarTap: React.PropTypes.func,
  canDeleteProject: React.PropTypes.bool,
  canDeleteTask: React.PropTypes.bool,
  canMove: React.PropTypes.bool,
  onDelete: React.PropTypes.func,
  onItemTap: React.PropTypes.func,
  onItemUpdate: React.PropTypes.func,
  onItemMove: React.PropTypes.func,
  style: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
  className: React.PropTypes.string,
};

export default ItemList;
