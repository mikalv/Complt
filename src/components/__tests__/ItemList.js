import React from 'react';
import { mount } from 'enzyme';
import ItemList from '../ItemList';

const task = { name: 'Task', isProject: false, isCompleted: false };
const project = { name: 'Project', isProject: true };

describe('ItemList component', () => {
  it('renders without crashing with no items', () => {
    mount(<ItemList />);
  });
  it('renders without crashing when an item is null', () => {
    mount(<ItemList items={[null]} />);
  });
  it('renders an item correctly', () => {
    const component = mount(<ItemList canMove items={[task]} />);
    expect(component).toMatchSnapshot();
  });
  it('calls onDelete correctly with a task', () => {
    const onDelete = jest.fn();
    const component = mount(<ItemList
      canDeleteTask
      onDelete={onDelete}
      items={[project, task, project]}
    />);
    component.childAt(1).find('MdDelete').simulate('click');
    expect(onDelete).toBeCalledWith(1);
  });
  it('calls onDelete correctly with a project', () => {
    const onDelete = jest.fn();
    const component = mount(<ItemList
      canDeleteProject
      onDelete={onDelete}
      items={[project, task, project]}
    />);
    component.childAt(2).find('MdDelete').simulate('click');
    expect(onDelete).toBeCalledWith(2);
  });
  it('calls onItemTap correctly', () => {
    const onItemTap = jest.fn();
    const component = mount(<ItemList
      onItemTap={onItemTap}
      items={[project, task, project]}
    />);
    component.childAt(2).find('ListItemText').simulate('click');
    expect(onItemTap).toBeCalledWith(2);
  });
  it('calls onItemAvatarTap correctly', () => {
    const onItemAvatarTap = jest.fn();
    const component = mount(<ItemList
      onItemAvatarTap={onItemAvatarTap}
      items={[project, task, project]}
    />);
    component.childAt(2).find('Avatar').simulate('click');
    expect(onItemAvatarTap).toBeCalledWith(2);
  });
  it('calls onItemUpdate correctly', () => {
    const onItemUpdate = jest.fn();
    const component = mount(<ItemList
      onItemUpdate={onItemUpdate}
      items={[project, task, project]}
    />);
    component.childAt(2).find('MdCreate').simulate('click');
    expect(onItemUpdate).toBeCalledWith(2);
  });
  it('calls onItemMove correctly', () => {
    const onItemMove = jest.fn();
    const component = mount(<ItemList
      onItemMove={onItemMove}
      canMove
      items={[project, task, project]}
    />);
    component.childAt(2).find('MdMoreVert').simulate('click');
    expect(onItemMove).toBeCalledWith(2);
  });
});
