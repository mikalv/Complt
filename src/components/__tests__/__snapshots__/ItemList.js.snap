exports[`ItemList component passes the canDelete prop to the Item component if the item is a project and the component has the canDeleteProject prop 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }>
  <div>
    <span
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}>
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 56,
              "paddingTop": 20,
              "position": "relative",
            }
          }>
          Project
          <div
            className={undefined}
            color={undefined}
            onTouchTap={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24">
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z" />
            </svg>
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 4,
                "transform": "translate(0, 0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button">
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}>
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24">
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z" />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ItemList component passes the canDelete prop to the Item component if the item is a task and the component has the canDeleteTask prop 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }>
  <div>
    <span
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}>
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 56,
              "paddingTop": 20,
              "position": "relative",
            }
          }>
          Task
          <div
            className={undefined}
            color={undefined}
            onTouchTap={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24">
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
            </svg>
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 4,
                "transform": "translate(0, 0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button">
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}>
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24">
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z" />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ItemList component passes the onItemAvatarTap function to the Item component 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }>
  <div>
    <span
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}>
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
            }
          }>
          Task
          <div
            className={undefined}
            color={undefined}
            onTouchTap={[Function]}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24">
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
            </svg>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ItemList component passes the onItemTap function to the Item component 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }>
  <div>
    <span
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}>
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
            }
          }>
          Task
          <div
            className={undefined}
            color={undefined}
            onTouchTap={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24">
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
            </svg>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ItemList component renders an item correctly 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }>
  <div>
    <span
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}>
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
            }
          }>
          Task
          <div
            className={undefined}
            color={undefined}
            onTouchTap={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24">
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
            </svg>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ItemList component renders without crashing when an item is null 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  } />
`;

exports[`ItemList component renders without crashing with no items 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  } />
`;
